float32 axis1
float32 axis2
float32 axis3
int32 btn1
int32 btn2
int32 btn3